src/nimble.nim

```diff
@@ -1480,7 +1480,7 @@ proc validateDevModeDepsWorkingCopiesBeforeLock(
     }
   
   # Remove not errors from the errors set.
-  for name, error in common.dup(errors):
+  for name, error in errors:
```

Comment:
`dup` was used here, because the code is deleting items from the collection while iterating.